specs/date_range_spec.rb

```diff
@@ -0,0 +1,20 @@
+require_relative 'spec_helper'
+
+describe "DateRange" do
+
+  let(:date) { BookingSystem::DateRange.new('2001-02-05', '2001-02-10') }
+  let(:hotel) { BookingSystem::Hotel.new }
+
+  describe "#initialize" do
+    it "should create an instance of DateRange" do
+      date.must_be_instance_of BookingSystem::DateRange
+
+      date.must_respond_to :check_in
+      date.check_in.must_be_instance_of Date
+
+      date.must_respond_to :check_out
+      date.check_out.must_be_instance_of Date
+    end
```

Comment:
You should have some tests for `overlap?` here! There are a lot of interesting cases around whether or not two date ranges overlap. My full list includes:
- Does overlap:
  - Same dates
  - Overlaps in the front
  - Overlaps in the back
  - Completely contained
  - Completely containing
- Does not overlap:
  - Completely before
  - Completely after
  - Ends on the checkin date
  - Starts on the checkout date